components/focus-trap/focus-trap-mixin.js

```diff
@@ -0,0 +1,99 @@
+import { getNextFocusable, getPreviousFocusable } from '../../helpers/focus.js';
+import { html } from 'lit-element/lit-element.js';
+import { isComposedAncestor } from '../../helpers/dom.js';
+
+export const FocusTrapMixin = superclass => class extends superclass {
+
+	static get properties() {
+		return {
+			trap: { type: Boolean, reflect: true }
```

Comment:
What the use case for the attribute? Like folks who want a focus trap, but don't want it to always be turned on?

Reply:
Dialogs is a great example - if we assume it's always trapping and we have a `d2l-dialog` in the DOM, it would be wired up to capture focus events elsewhere in the body (see the body event handler, which redirects focus to the trap) even when the dialog is not open.

Reply:
I see. So dialog would bind the `trap` attribute to its own `opened`.

Reply:
An alternative is to have dialog conditionally render `d2l-focus-trap` based on it's open state.  Not sure how I feel about that though ... not sure I like the dialog internal DOM changing to that degree when the dialog is opened/closed.

Reply:
Yeah agreed that sounds a little wonky. The other option would be for the focus trap to know whether or not it's actually visible in the DOM, but that also sounds complicated.